src/embedding.rs

```diff
@@ -191,6 +277,88 @@ impl MatrixWrapper for MMapMatrix {
         }
     }
 
+    fn init_with_evec(
+        rows: usize,
+        cols: usize,
+        sparse_matrix_reader: Arc<SparseMatrix>,
+        order: TruncatedOrder
+    ) -> Self {
+        let uuid = Uuid::new_v4();
+        let file_name = format!("{}_matrix_{}", sparse_matrix_reader.get_id(), uuid);
+        let mut mmap = create_mmap(rows, cols, file_name.as_str());
+
+        let matrix_transform = |x: ArrayView2<f32>| -> Array2<f32> {
+            let shape = x.shape();
+            let mut arr = Array2::zeros((shape[0], shape[1])); 
+    
+            let sparse_laplacian_entries: Vec<Entry> = sparse_matrix_reader.entries.clone() // Might be a problem?
+            .into_par_iter()
+            .map(|entry| {
+                if entry.col == entry.row { // What happens for reflexive columns?
+                    Entry {
+                        row: entry.row,
+                        col: entry.col,
+                        value: sparse_matrix_reader.get_row_sum(entry.row),
```

Comment:
Is it correct?

It seems weird to me that we take row_sums in a iterator over edges. So we might consider some sums many times. Is it working as expected?

Reply:
You're right, that was supposed to add the diagonal entries, but this obviously doesn't work as expected.